File: Source/Eidolune/Core/Trigger.cpp


#include "Trigger.h"

std::string Trigger::ToString() const { return ScriptReference; }

Trigger::Trigger(const json& j) {
    ScriptReference = j["script_reference"];
    Description = j["description"];
    Event = j.contains("event") ? j["event"] : "";
}


std::string Trigger::GetName() const {
    return ScriptReference;
}

File: Backend/core/models/trigger.py
from django.db import models

class Trigger(models.Model):
    script_reference = models.CharField(max_length=100)
    description = models.TextField(blank=True)

    event = models.CharField(max_length=100, default="")

    
    def __str__(self):
        return self.script_reference
